tests/unit/rdf/test_resource_serialisation.py

```diff
@@ -314,6 +315,175 @@ class A(NewResource):
         URIRef("http://some-entity-uri"),
     ) in graph
 
+def test_markdown_datatype():
+    """ """
+
+    class A(NewResource):
+        p: Annotated[
+            str,
+            Triple(
+                URIRef("http://original-uri"),
+                PropertyStatus.recommended,
+                map_str_to_markdown,
+            ),
+        ]
+    a = A("http://some-a-uri")
+    a.p = "Hello, World"
+
+    graph = a.to_graph(Graph())
+
+    assert len(graph) == 2
+    assert (
+        URIRef("http://some-a-uri"),
+        URIRef("http://original-uri"),
+        Literal("Hello, World", datatype="https://www.w3.org/ns/iana/media-types/text/markdown#Resource"),
+    ) in graph
+
+    assert (
+        URIRef("http://some-a-uri"),
+        RDF.type,
+        RDFS.Resource,
+    ) in graph
+
+def test_italic_detected_as_html(capsys):
+    """Testing italicized text in a markdown isn't detected as html."""
+    example_html_in_markdown = """
+    *italicized text*
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out + captured.err == ""
+
+def test_link_detected_as_html(capsys):
+    """Testing a link in a markdown isn't detected as html."""
+    example_html_in_markdown = """
+    [title](https://www.example.com)
+    """
+    map_str_to_markdown(example_html_in_markdown)
```

Comment:
A more reliable way to test this would be to refactor the logic which looks for HTML out of `map_str_to_markdown(...)` into a separate function which just returns a bool saying yes/no to the presence of HTML.

You could then test that function's result instead of looking at the log (which as you discovered can be mis-configured).